calicoctl/commands/node/status.go

```diff
@@ -129,21 +132,41 @@ func printBGPPeers(ipv string) {
 		os.Exit(1)
 	}
 
-	buf := make([]byte, 1024)
+	// BIRD socket read timeout.
+	timeOut := 2 * time.Second
+	bufReader := bufio.NewReader(c)
 
-	n, err := c.Read(buf[:])
-	if err != nil {
-		fmt.Printf("Error: %v\n", err)
+	birdOut := []string{}
+
+	for {
+		// Set a time-out for reading from the socket connection.
+		// Read operation will fail if no data is received until the timeout.
+		c.SetReadDeadline(time.Now().Add(timeOut))
+
+		// Read string with \n as delim.
+		str, err := bufReader.ReadString('\n')
+		if err != nil {
+			fmt.Println(err)
+			break
+		}
+
+		log.Debugf("Reading line from BIRD sock: %s\n", str)
```

Comment:
Shorted to "Read: %s\n"

And maybe put a debug log outside the for loop saying that you're reading from the BIRD socket, and put some debug trace in the branch where you exit the loop and some trace after you exit the loop.